mediator/esmFldsExchange_coastal_mod.F90

```diff
@@ -236,6 +258,83 @@ subroutine esmFldsExchange_coastal_advt(gcomp, phase, rc)
       deallocate(S_flds)
     end if
 
+    ! ---------------------------------------------------------------------
+    ! to ice: atm fields
+    ! ---------------------------------------------------------------------
+    if (coastal_attr%atm_present .and. coastal_attr%ice_present) then
+      allocate(S_flds(14))
+      S_flds = (/'Sa_u10m',       & ! inst_zonal_wind_height10m
+                 'Sa_v10m',       & ! inst_merid_wind_height10m                                     
+                 'Sa_z',       & ! lowest atmospheric height                                     
+                 'Sa_t2m',    & ! tenperature at lowest atm level                               
+                 'Sa_q2m',    & ! Specific Humidity                                             
+                 'Sa_pslv',    & ! pressure bottom  Faxa_rain',  & ! Rain
+                 'Faxa_snow',  & ! Snow
+                 'Faxa_swvdr', & ! Short-wave
+                 'Faxa_swvdf', & ! Short-wave
+                 'Faxa_swndr', & ! Short-wave
+                 'Faxa_swndf', & ! Short-wave
+                 'Faxa_swnet', & ! Short-wave net
+                 'Faxa_lwdn' /) ! long-wave
+                 
+      do n = 1,size(S_flds)
+         fldname = trim(S_flds(n))
+         call addfld_from(compatm, trim(fldname))
+         call addfld_to(compice, trim(fldname))
+      end do
+      deallocate(S_flds)
+    end if
+
+    ! ---------------------------------------------------------------------
+    ! to ice: ocean fields 
+    ! ---------------------------------------------------------------------
+    if (coastal_attr%ocn_present .and. coastal_attr%ice_present) then
```

Comment:
@SmithJos13 Same in here. Please make sure to use same length for strings in the array.